src/App.vue

```diff
@@ -239,67 +374,74 @@ export default {
-        { icon: 'share', title: 'Network graph', path: '/mesh' }
-      ],
-      settings: new Settings(localStorage),
-      status: '',
-      statusColor: '',
-      drawer: false,
-      mini: false,
-      topbar: [],
-      title: '',
-      snackbar: false,
-      snackbarText: '',
-      dark: undefined,
-      baseURI: ConfigApis.getBasePath()
-    }
-  },
-  watch: {
-    $route: function (value) {
-      this.title = value.name || ''
     },
-    dark (v) {
-      this.settings.store('dark', this.dark)
+    async startSocket () {
+      if (this.socket || !this.$route.meta || !this.$route.meta.requiresAuth) {
+        return
+      }
 
-      this.$vuetify.theme.dark = v
-      this.changeThemeColor()
-    }
-  },
-  beforeMount () {
-    this.title = this.$route.name || ''
+      if (!this.user || !this.user.token) {
+        await this.logout()
+      }
 
-    const self = this
+      this.socket = io('/', {
+        path: ConfigApis.getSocketPath(),
+        query: { token: this.user.token }
+      })
 
-    this.socket = io('/', {
-      path: ConfigApis.getSocketPath()
-    })
+      this.socket.on('connect', () => {
+        this.updateStatus('Connected', 'green')
+      })
 
-    this.socket.on('connect', () => {
-      self.updateStatus('Connected', 'green')
-    })
+      this.socket.on('disconnect', () => {
+        this.updateStatus('Disconnected', 'red')
+      })
 
-    this.socket.on('disconnect', () => {
-      self.updateStatus('Disconnected', 'red')
-    })
+      this.socket.on('error', () => {
+        console.log('Socket error')
+      })
 
-    this.socket.on('error', () => {
-      console.log('Socket error')
-    })
+      this.socket.on('reconnecting', () => {
+        this.updateStatus('Reconnecting', 'yellow')
+      })
 
-    this.socket.on('reconnecting', () => {
-      self.updateStatus('Reconnecting', 'yellow')
-    })
+      this.socket.on(socketEvents.init, data => {
+        // convert node values in array
+        this.initNodes(data.nodes)
+        this.setControllerStatus(data.error ? data.error : data.cntStatus)
+        this.setAppInfo(data.info)
+      })
+
+      this.socket.on(socketEvents.connected, this.setAppInfo.bind(this))
+      this.socket.on(
+        socketEvents.controller,
+        this.setControllerStatus.bind(this)
+      )
+
+      this.socket.on(socketEvents.nodeUpdated, this.initNode.bind(this))
+      this.socket.on(socketEvents.nodeRemoved, this.initNode.bind(this))
+
+      this.socket.on(socketEvents.valueRemoved, this.removeValue.bind(this))
+      this.socket.on(socketEvents.valueUpdated, this.updateValue.bind(this))
+
+      this.socket.emit(socketActions.init, true)
+    },
+    async logout () {
+      const user = Object.assign({}, this.user)
+      localStorage.setItem('user', JSON.stringify(user))
+      localStorage.removeItem('logged')
+
+      try {
+        await ConfigApis.logout()
+      } catch (error) {}
```

Comment:
might want to tell the user the logout failed?

Reply:
in general when user logs out everything is deleted from local storage so even if that request fails he needs to re-login, btw I will add an error here to make this more clear